src/runtime/cuda/cuda_device_api.cc

```diff
@@ -241,5 +242,38 @@ TVM_REGISTER_GLOBAL("device_api.cpu_pinned").set_body([](TVMArgs args, TVMRetVal
   *rv = static_cast<void*>(ptr);
 });
 
+class GPUTimerNode : public TimerNode {
+ public:
+  virtual void Start() { cudaEventRecord(start_, CUDAThreadEntry::ThreadLocal()->stream); }
```

Comment:
We may wrap it as `CUDA_CALL(cudaEventRecord(start_, CUDAThreadEntry::ThreadLocal()->stream));`. Same for other cuda APIs.